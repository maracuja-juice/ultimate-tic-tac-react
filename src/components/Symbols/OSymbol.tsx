import * as React from 'react';
import './symbols.css';
import './symbolAnimations.css';
import '../../fonts/icons.css';
import * as classNames from 'classnames';

interface OSymbolProps {
  bigSymbol?: boolean;
  shouldAnimate?: boolean;
  className?: string;
}

export class OSymbol extends React.Component<OSymbolProps> {
  public static defaultProps: Partial<OSymbolProps> = {
    shouldAnimate: true,
  };

  render() {
    const { bigSymbol, shouldAnimate, className } = this.props;
    const iconClass = classNames({
      'icon-o': true,
      o: true,
      'big-symbol': bigSymbol,
      'animate-o': shouldAnimate,
      className: className,
    });

    return <i className={iconClass} aria-label="Circle Symbol" />;
  }
}
