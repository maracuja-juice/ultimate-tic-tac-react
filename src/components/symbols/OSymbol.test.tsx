import * as React from 'react';
import * as Enzyme from 'enzyme';
import { OSymbol } from './OSymbol';
import { shallowToJson } from 'enzyme-to-json';
import * as Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

describe( 'OSymbol', () => {
    it('matches its snapshot', () => {
        Enzyme.configure({ adapter: new Adapter() });

        const osymbol = shallow(<OSymbol/>);
        expect(shallowToJson(osymbol)).toMatchSnapshot();
    });
});