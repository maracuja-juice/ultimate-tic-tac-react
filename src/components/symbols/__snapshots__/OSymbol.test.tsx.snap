// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSymbol matches its snapshot 1`] = `
<i
  className="material-icons o  animate-o"
>
  panorama_fish_eye
</i>
`;
